srcts/src/shiny/bind.ts

```diff
@@ -204,10 +223,17 @@ function bindInputs(
       if (el.hasAttribute("data-shiny-no-bind-input")) continue;
       const id = binding.getId(el);
 
-      // Check if ID is falsy, skip
+      // Don't bind if ID is falsy...
       if (!id) continue;
 
-      bindingsRegistery.addBinding(id, "input");
+      // ...or if this element is already registered with a binding
+      if ($(el).data("shiny-input-binding")) {
+        if (!bindingsRegistry.isRegistered(id, "input")) {
+          bindingsRegistry.addBinding(id, "input");
+        }
```

Comment:
As this is the only use of this method and it immediately follows logic involving the registry itself, I think it should be kept internal and added to the logic of `addBinding()` in order to keep the exposure to the surface of the registry implementation as small as possible.

Reply:
thanks, yeah I took this out in b67b0623e9f8cb65175a89a369f2191e6779b838